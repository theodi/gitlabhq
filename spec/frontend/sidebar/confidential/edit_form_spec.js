import { shallowMount } from '@vue/test-utils';
import EditForm from '~/sidebar/components/confidential/edit_form.vue';

describe('Edit Form Dropdown', () => {
  let wrapper;
  const toggleForm = () => {};
  const updateConfidentialAttribute = () => {};

  const createComponent = props => {
    wrapper = shallowMount(EditForm, {
      propsData: {
        ...props,
        isLoading: false,
        fullPath: '',
        issuableType: 'issue',
      },
    });
  };

  afterEach(() => {
    wrapper.destroy();
    wrapper = null;
  });

  describe('when not confidential', () => {
    it('renders "You are going to turn on the confidentiality." in the ', () => {
      createComponent({
        confidential: false,
        toggleForm,
        updateConfidentialAttribute,
      });

      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('when confidential', () => {
    it('renders on or off text based on confidentiality', () => {
      createComponent({
        confidential: true,
        toggleForm,
        updateConfidentialAttribute,
      });

      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
