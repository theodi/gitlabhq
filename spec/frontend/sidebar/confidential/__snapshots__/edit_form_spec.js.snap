// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit Form Dropdown when confidential renders on or off text based on confidentiality 1`] = `
<div
  class="dropdown show"
  toggleform="function () {}"
  updateconfidentialattribute="function () {}"
>
  <div
    class="dropdown-menu sidebar-item-warning-message"
  >
    <div>
      <p>
        <gl-sprintf-stub
          message="You are going to turn off the confidentiality. This means %{strongStart}everyone%{strongEnd} will be able to see and leave a comment on this %{issuableType}."
        />
      </p>
       
      <edit-form-buttons-stub
        confidential="true"
        fullpath=""
      />
    </div>
  </div>
</div>
`;

exports[`Edit Form Dropdown when not confidential renders "You are going to turn on the confidentiality." in the  1`] = `
<div
  class="dropdown show"
  toggleform="function () {}"
  updateconfidentialattribute="function () {}"
>
  <div
    class="dropdown-menu sidebar-item-warning-message"
  >
    <div>
      <p>
        <gl-sprintf-stub
          message="You are going to turn on the confidentiality. This means that only team members with %{strongStart}at least Reporter access%{strongEnd} are able to see and leave comments on the %{issuableType}."
        />
      </p>
       
      <edit-form-buttons-stub
        fullpath=""
      />
    </div>
  </div>
</div>
`;
